#!/bin/bash
#SBATCH --job-name=DADA_ASV
#SBATCH --cpus-per-task=24
#SBATCH --mem=128G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=meier@microbial-ecology.net
#SBATCH --output=DADADA.out
#SBATCH --error=DADADA.err
#SBATCH --nice=1000
#SBATCH --partition=himem

DATA_FOLDER=/scratch/dmeier/Avdat_amplicons
DADA_SCRIPT_LOCATION=/proj/dmeier/scripts

module load R

cd $TMPDIR

#copy the fastq files to tmp
scp $DATA_FOLDER/reads/*merged.fastq.gz ./

#copy the R-script to temp
scp $DADA_SCRIPT_LOCATION/Dada2.R ./

#run Dada2 on whatever sequences are in the tmp
Rscript Dada2.R

#fix the column names in the table and write the he OTU table back to scratch, including sequences, generated by Dada2
sed 's/_merged.fastq.gz//g' ASV_table_nochim.tsv > $DATA_FOLDER/ASV_table_nochim.tsv

#generate a fasta file with ASV sequences
cut -f 1,2 ASV_table_nochim.tsv | sed '1d' | perl -0pe 's/ASV_/>ASV_/g' | perl -0pe 's/\t/\n/g' > $DATA_FOLDER/ASV_sequences.fasta

#delete the temp
rm -r /tmp/dmeier
