#!/bin/bash
#SBATCH --job-name=Emapper
#SBATCH --cpus-per-task=1
#SBATCH --mem=8G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=your.email@provider.net
#SBATCH --output=logs/Emapper.%a.out
#SBATCH --error=logs/Emapper.%a.err
#SBATCH --nice=1000
#SBATCH --partition=basic
#SBATCH --constraint=array-1core&fast-localmirror


DATA_FOLDER=/scratch/dmeier/Avdat_metagenomes

#copy input files to tmp. This files are generated by splitting the innitial big file with all sequences
scp $DATA_FOLDER/Final_bins/Final_bins.$SLURM_ARRAY_TASK_ID.faa $TMPDIR/

mkdir $TMPDIR/Emapper_Diamond
mkdir $DATA_FOLDER/Final_Bins/Emapper_Diamond

#load emapper module
module load eggnogmapper

#Diamond-based search
emapper.py \
    --cpu 1 \
    -m diamond \
    --data_dir /localmirror/monthly/mapper-data/ \
    --dmnd_db /localmirror/monthly/mapper-data/eggnog_proteins.dmnd \
    --qtype seq \
    --seed_ortholog_evalue 0.00001 \
    -i $TMPDIR/Final_bins.$SLURM_ARRAY_TASK_ID.faa \
    --output_dir $TMPDIR/Emapper_Diamond \
    -o Final_bins_EggNOG.$SLURM_ARRAY_TASK_ID

#copy results back to scratch
    scp Emapper_Diamond/* $DATA_FOLDER/Final_Bins/Emapper_Diamond/

#delete temporary directory
rm -r /tmp/dmeier_emapper$SLURM_ARRAY_TASK_ID
